src/necessities.md

```diff
@@ -109,3 +109,42 @@ cannot be used in some situations where most of the Rust runtime stack can.
 The license of a crate's dependencies can affect the restrictions on
 distribution of the crate itself, so a permissively-licensed crate should
 generally only depend on permissively-licensed crates.
+
+<a id="c-msrv"></a>
+## Crate has an MSRV policy (C-MSRV)
+
+A crate should clearly document its Minimal Supported Rust Version:
+
+* Which versions versions of Rust are supported now?
+* Under what conditions is MSRV increased?
+* How are MSRV increases reflected in the semver version of the crate?
+
+Compliance with a crate’s stated MSRV should be tested in CI.
+
+The API guidelines tentatively suggest that, for libraries, an MSRV increase should
+*not* be considered a semver-breaking change. Specifically, when increasing
+MSRV:
+
+* for crates past `1.0.0`, increment minor version (`1.1.3` -> `1.2.0`),
+* for crates before `1.0.0`, increment patch version (`0.1.3` -> `0.1.4`).
+
+This reduces the amount of ecosystem-wide work for MSRV upgrades and prevents
+incompatibilities. It also is a de-facto practice for many cornerstone crates.
+This policy gives more power to library consumers to manually select working
+combinations of library and compiler versions, at the cost of breaking `cargo
+update` workflow for older compilers.
+
+However, do not increase MSRV without a good reason, and, if possible, batch
+MSRV increases with semver-breaking changes.
+
+Nonetheless, some crates intentionally choose to treat MSRV increases as a semver
+breaking change. This is also a valid strategy, but it is not recommended as the
+default choice.
+
+To reliably test MSRV on CI, use a dedicated `Cargo.lock` file with dependencies
+pinned to minimal versions:
+
+```bash
+$ cp ci/Cargo.lock.min ./Cargo.lock
+$ cargo +$MSRV build
```

Comment:
No need to do a full build.
```suggestion
$ cargo +$MSRV check
```

Reply:
Well, actually I guess this breaks if you have a *really* old build. Maybe this?
```suggestion
$ cargo +$MSRV rustc -- --profile check
```

Reply:
Note that even if `cargo check` succeeds, the build may fail...

Reply:
It could? When would that happen, barring a compiler bug?

Reply:
Yeah, see rust-lang/rust#49292, rust-lang/rust#61330, rust-lang/rust#70923, etc.

> `cargo check` performs all steps up to translation. Some errors are only detected during later phases.

Reply:
I think those are considered bugs. https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AI-monomorphization

Reply:
You can't also exclude some weird linking error due to some c library doing autocfg-style version sniffing. 

Generally, as we are not running or compiling tests, `cargo build` on MSRV is already bound to finish faster than any other CI job, so, in my mind, it makes little sense to micro-optimize here.